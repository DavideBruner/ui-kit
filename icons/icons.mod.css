@import '../materials/colors.mod.css';

.theme-black {
  fill: var(--new-color-black);
}

.theme-grey {
  fill: var(--new-color-gray-60);
}

.theme-white {
  fill: var(--new-color-white);
}

.theme-blue {
  fill: var(--new-color-blue);
}

.theme-green {
  fill: var(--new-color-green-25);
}

.theme-apple-green {
  fill: var(--new-color-green);
}

.theme-orange {
  fill: var(--new-color-orange);
}

.theme-red {
  fill: var(--new-color-red);
}

svg.scale:not(:root) {
  width: 100%;
  height: auto;
}
