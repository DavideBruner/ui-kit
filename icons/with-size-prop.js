import { mapProps } from 'recompose';
import classnames from 'classnames';
import styles from './icons.mod.css';

export default mapProps(({ size, ...remainingProps }) => ({
  ...remainingProps,
  ...(size === 'scale'
    ? { className: classnames(remainingProps.className, styles.scale) }
    : {}),
  width: size === 'small' ? 16 : 24,
  height: size === 'small' ? 16 : 24,
}));
