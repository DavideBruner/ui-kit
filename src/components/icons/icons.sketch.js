import React from 'react';
import styled from '@emotion/styled';
import * as UIKit from 'ui-kit';
import { Suite, SketchSpec } from '../../../test/percy';

const Inline = styled.div`
  display: flex;
  flex-direction: row;
`;

const icons = Object.keys(UIKit).filter(thing => thing.endsWith('Icon'));

const sizes = ['small', 'medium', 'big', 'scale'];
const themes = [
  'black',
  'grey',
  'white',
  'blue',
  'green',
  'green-light',
  'orange',
  'red',
];

export default (
  <Suite>
    <SketchSpec label="Icons">
      {icons.map(iconName => {
        const Icon = UIKit[iconName];
        return sizes.map(size => (
          <div key={`${iconName}-${size}`}>
            <Inline>
              {themes.map(theme => (
                <div
                  key={theme}
                  data-sketch-symbol={`Icon/${iconName}/${size}/${theme}`}
                >
                  <Icon size={size} theme={theme} />
                </div>
              ))}
            </Inline>
          </div>
        ));
      })}
    </SketchSpec>
  </Suite>
);
