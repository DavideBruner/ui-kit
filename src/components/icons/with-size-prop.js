import { mapProps } from 'recompose';
import classnames from 'classnames';
import styles from './icons.mod.css';

const ICON_SIZES = {
  SMALL: 12,
  MEDIUM: 16,
  BIG: 24,
};

const getIconSize = size => {
  switch (size) {
    case 'small':
      return ICON_SIZES.SMALL;
    case 'medium':
      return ICON_SIZES.MEDIUM;
    case 'big':
      return ICON_SIZES.BIG;
    default:
      return ICON_SIZES.BIG;
  }
};

export default mapProps(({ size, ...remainingProps }) => ({
  ...remainingProps,
  ...(size === 'scale'
    ? { className: classnames(remainingProps.className, styles.scale) }
    : {}),
  width: getIconSize(size),
  height: getIconSize(size),
}));
