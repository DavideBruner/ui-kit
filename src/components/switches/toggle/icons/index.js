import ToggleButtonIcon from './toggle-button-default.svg';
import ToggleBarIcon from './toggle-bar-default.svg';

export default {
  ButtonDefault: ToggleButtonIcon,
  BarDefault: ToggleBarIcon,
};
