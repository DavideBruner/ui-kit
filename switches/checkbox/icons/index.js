// NOTE: explicit names for component names
import CheckboxDefaultIcon from './checkbox-default.svg';
import CheckboxDefaultHoverIcon from './checkbox-hover.svg';
import CheckboxDefaultErrorIcon from './checkbox-error.svg';
import CheckboxCheckedIcon from './checkbox-checked.svg';
import CheckboxCheckedHoverIcon from './checkbox-checked-hover.svg';
import CheckboxCheckedErrorIcon from './checkbox-checked-error.svg';
import CheckboxDisabledIcon from './checkbox-disabled.svg';
import CheckboxCheckedDisabledIcon from './checkbox-checked-disabled.svg';
import CheckboxIndeterminateIcon from './checkbox-indeterminate.svg';
import CheckboxIndeterminateHoverIcon from './checkbox-indeterminate-hover.svg';
import CheckboxIndeterminateDisabledIcon from './checkbox-indeterminate-disabled.svg';
import CheckboxIndeterminateErrorIcon from './checkbox-indeterminate-error.svg';

export default {
  Default: CheckboxDefaultIcon,
  Hover: CheckboxDefaultHoverIcon,
  Error: CheckboxDefaultErrorIcon,
  Checked: CheckboxCheckedIcon,
  CheckedHover: CheckboxCheckedHoverIcon,
  CheckedError: CheckboxCheckedErrorIcon,
  Disabled: CheckboxDisabledIcon,
  CheckedDisabled: CheckboxCheckedDisabledIcon,
  Indeterminate: CheckboxIndeterminateIcon,
  IndeterminateHover: CheckboxIndeterminateHoverIcon,
  IndeterminateDisabled: CheckboxIndeterminateDisabledIcon,
  IndeterminateError: CheckboxIndeterminateErrorIcon,
};
