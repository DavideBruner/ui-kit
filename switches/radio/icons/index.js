// NOTE: explicit names for component names
import RadioOptionDefaultIcon from './radio-option-default.svg';
import RadioOptionDefaultHoverIcon from './radio-option-hover.svg';
import RadioOptionCheckedIcon from './radio-option-checked.svg';
import RadioOptionCheckedHoverIcon from './radio-option-checked-hover.svg';
import RadioOptionDisabledIcon from './radio-option-disabled.svg';
import RadioOptionCheckedDisabledIcon from './radio-option-checked-disabled.svg';

export default {
  Default: RadioOptionDefaultIcon,
  Hover: RadioOptionDefaultHoverIcon,
  Checked: RadioOptionCheckedIcon,
  CheckedHover: RadioOptionCheckedHoverIcon,
  Disabled: RadioOptionDisabledIcon,
  CheckedDisabled: RadioOptionCheckedDisabledIcon,
};
