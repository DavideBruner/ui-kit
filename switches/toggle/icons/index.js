import withSizeProp from '../with-size-prop';
import ToggleBackgroundDefaultIcon from './toggle-normal-bkgr.svg';
import ToggleBackgroundHoverIcon from './toggle-active-hover-bkgr.svg';
import ToggleBackgroundDisabledIcon from './toggle-disable-bkgr.svg';
import ToggleBackgroundCheckedIcon from './toggled-bkgr.svg';
import ToggleBackgroundCheckedHoverIcon from './toggled-active-hover-bkgr.svg';
import ToggleBackgroundCheckedDisabledIcon from './toggled-disable-bkgr.svg';
import ToggleButtonDefaultIcon from './toggle-normal-bt.svg';
import ToggleButtonHoverIcon from './toggle-active-hover-bt.svg';
import ToggleButtonDisabledIcon from './toggle-disable-bt.svg';
import ToggleButtonCheckedIcon from './toggled-bt.svg';
import ToggleButtonCheckedHoverIcon from './toggled-active-hover-bt.svg';
import ToggleButtonCheckedDisabledIcon from './toggled-disable-bt.svg';

export default {
  BackgroundDefault: ToggleBackgroundDefaultIcon,
  BackgroundHover: ToggleBackgroundHoverIcon,
  BackgroundDisabled: ToggleBackgroundDisabledIcon,
  BackgroundChecked: ToggleBackgroundCheckedIcon,
  BackgroundCheckedHover: ToggleBackgroundCheckedHoverIcon,
  BackgroundCheckedDisabled: ToggleBackgroundCheckedDisabledIcon,
  ButtonDefault: withSizeProp(ToggleButtonDefaultIcon),
  ButtonHover: withSizeProp(ToggleButtonHoverIcon),
  ButtonDisabled: withSizeProp(ToggleButtonDisabledIcon),
  ButtonChecked: withSizeProp(ToggleButtonCheckedIcon),
  ButtonCheckedHover: withSizeProp(ToggleButtonCheckedHoverIcon),
  ButtonCheckedDisabled: withSizeProp(ToggleButtonCheckedDisabledIcon),
};
