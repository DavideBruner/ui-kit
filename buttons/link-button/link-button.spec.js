import React from 'react';
import { shallow } from 'enzyme';
import Text from '../../typography/text';
import { AddIcon } from '../../icons';
import LinkButton from './link-button';

const createTestProps = props => ({
  label: 'Accessibility text',
  to: '/foo/bar',
  iconLeft: <AddIcon size="medium" />,
  isDisabled: false,
  ...props,
});

describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<LinkButton {...props} />);
  });

  it('outputs correct tree', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <Link>', () => {
    expect(wrapper).toRender('Link');
  });

  it('should pass the url to <Link>', () => {
    expect(wrapper.find('Link')).toHaveProp('to', '/foo/bar');
  });

  it('should render a <Text.Body>', () => {
    expect(
      wrapper.containsMatchingElement(
        <Text.Body>{'Accessibility text'}</Text.Body>
      )
    ).toBe(true);
  });

  it('should render the icon', () => {
    expect(wrapper).toRender('AddIcon');
  });

  describe('with data-* props', () => {
    beforeEach(() => {
      props = createTestProps({
        'data-track-component': 'LinkButton',
        'data-track-label': 'LinkButton',
        'data-track-event': 'click',
        'data-test': 'link-button',
      });
      wrapper = shallow(<LinkButton {...props} />);
    });
    it('should apply given `data-track-component` to LinkButton', () => {
      expect(wrapper.find('Link')).toHaveProp(
        'data-track-component',
        expect.stringMatching('LinkButton')
      );
    });
    it('should apply given `data-track-event` to AccessibleButton', () => {
      expect(wrapper.find('Link')).toHaveProp(
        'data-track-event',
        expect.stringMatching('click')
      );
    });
    it('should apply given `data-track-label` to AccessibleButton', () => {
      expect(wrapper.find('Link')).toHaveProp(
        'data-track-label',
        expect.stringMatching('LinkButton')
      );
    });
    it('should apply given `data-test` to AccessibleButton', () => {
      expect(wrapper.find('Link')).toHaveProp(
        'data-test',
        expect.stringMatching('link-button')
      );
    });
  });

  describe('without data-* props', () => {
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<LinkButton {...props} />);
    });
    it('should apply default `data-track-component` to LinkButton', () => {
      expect(wrapper.find('Link')).toHaveProp(
        'data-track-component',
        expect.stringMatching('LinkButton')
      );
    });
  });

  describe('if disabled', () => {
    beforeEach(() => {
      props = createTestProps({ isDisabled: true });
      wrapper = shallow(<LinkButton {...props} />);
    });
    it('renders a <Link> with disabled class', () => {
      expect(wrapper.find('Link')).toHaveClassName('disabled');
    });
    it('renders the icon with "grey" theme', () => {
      expect(wrapper.find('AddIcon')).toHaveProp('theme', 'grey');
    });
  });
});
