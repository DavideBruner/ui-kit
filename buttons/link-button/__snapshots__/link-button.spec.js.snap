// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering outputs correct tree 1`] = `
<Link
  className="button"
  replace={false}
  to="/foo/bar"
>
  <AddIcon
    size="small"
    theme="green"
  />
  <TextBody>
    Accessibility text
  </TextBody>
</Link>
`;
